test-data/unit/check-classes.test

```diff
@@ -2051,6 +2051,16 @@ class X: pass
 foo(X)
 [out]
 
+[case testTypeUsingTypeCTypeAnyMember]
+from typing import Type, Any
+def foo(arg: Type[Any]):
+    x = arg.member_name
```

Comment:
Can you also try an assignment to an arbitrary member? 

We could perhaps at least give a precise type for things like `arg.mro` which are defined in `type`. Maybe only fall back to `Any` if `type` doesn't have the attribute? If this is hard to do it's okay to leave it out from this PR.